src/sources/nats.rs

```diff
@@ -196,10 +203,15 @@ async fn nats_source(
                                 now,
                             );
 
+                            let legacy_subject_key_field = config
+                                .subject_key_field
+                                .path
+                                .as_ref()
+                                .map(LegacyKey::InsertIfEmpty);
```

Comment:
The schema has this as Overwrite.

Reply:
Updated in 2f79651bd, made both InsertIfEmpty.